src/manifolds/ProbabilitySimplex.jl

```diff
@@ -0,0 +1,371 @@
+after [`check_manifold_point`](@ref)`(M,p)`, `X` has to be of same dimension as `p`
+and its elements have to sum to one.
+The optional parameter `check_base_point` indicates, whether to call
+[`check_manifold_point`](@ref)  for `p` or not.
+The tolerance for the last test can be set using the `kwargs...`.
+"""
+function check_tangent_vector(
+    M::ProbabilitySimplex,
+    p,
+    X;
+    check_base_point = true,
+    kwargs...,
+)
+    if check_base_point
+        mpe = check_manifold_point(M, p; kwargs...)
+        mpe === nothing || return mpe
+    end
+    mpv = invoke(
+        check_tangent_vector,
+        Tuple{typeof(get_embedding(M)),typeof(p),typeof(X)},
+        get_embedding(M),
+        p,
+        X;
+        check_base_point = false, # already checked above
+        kwargs...,
+    )
+    mpv === nothing || return mpv
+    if !isapprox(sum(X), 0.0; kwargs...)
+        return DomainError(
+            sum(X),
+            "The vector $(X) is not a tangent vector to $(p) on $(M), since its elements to not sum up to 0.",
+        )
+    end
+    return nothing
+end
+
+decorated_manifold(M::ProbabilitySimplex) = Euclidean(representation_size(M)...; field = ℝ)
+
+default_metric_dispatch(::ProbabilitySimplex, ::FisherRaoMetric) = Val(true)
+
+@doc raw"""
+    distance(M,p,q)
+
+Compute the distance between two points on the [`ProbabilitySimplex`](@ref) `M`.
+The formula reads
+````math
+d_{Δ^n}(p,q) = 2\arccos \biggl( \sum_{i=1}^{n+1} \sqrt{p_i q_i} \biggr)
+````
+"""
+function distance(::ProbabilitySimplex, p, q)
+    sumsqrt = zero(Base.promote_eltype(p, q))
+    @inbounds for i in eachindex(p, q)
+        sumsqrt += sqrt(p[i] * q[i])
+    end
+    return 2 * acos(sumsqrt)
+end
+
+embed!(::ProbabilitySimplex, q, p) = copyto!(q, p)
+embed!(::ProbabilitySimplex, Y, p, X) = copyto!(Y, X)
+
+@doc raw"""
+    exp(M::ProbabilitySimplex,p,X)
+
+Compute the exponential map on the probability simplex.
+
+````math
+\exp_pX = \frac{1}{2}\Bigl(p+\frac{X_p^2}{\lVert X_p \rVert^2}\Bigr)
++ \frac{1}{2}\Bigl(p - \frac{X_p^2}{\lVert X_p \rVert^2}\Bigr)\cos(\lVert X_p\rVert)
++ \frac{1}{\lVert X_p \rVert}\sqrt{p}\sin(\lVert X_p\rVert),
+````
+
+where $X_p = \frac{X}{\sqrt{p}}$, with its division meant elementwise, as well as for the
+operations $X_p^2$ and $\sqrt{p}$.
+"""
+exp(::ProbabilitySimplex, ::Any...)
+
+function exp!(::ProbabilitySimplex, q, p, X)
+    s = sqrt.(p)
+    Xs = X ./ s ./ 2
+    θ = norm(Xs)
+    q .= (cos(θ) .* s .+ usinc(θ) .* Xs) .^ 2
+    return q
+end
+
+@doc raw"""
+    injectivity_radius(M,p)
+
+compute the injectivity radius on the [`ProbabilitySimplex`](@ref) `M` at the point `p`,
+i.e. the distanceradius to a point near/on the boundary, that could be reached by following the
+geodesic.
+"""
+function injectivity_radius(M::ProbabilitySimplex, p)
+    i = argmin(p)
+    q = copy(p)
+    q[i] = 0
+    q ./= sum(q)
+    return distance(M, p, q)
+end
```

Comment:
Combining with `distance` and then simplifying produces this much simpler one. Also by not calling `distance`, we don't imply that the boundary point is in the `ProbabilitySimplex`: 
```suggestion
function injectivity_radius(::ProbabilitySimplex{n}, p) where {n}
    i = argmin(p)
    s = sum(p) - p[i]
    return 2 * acos(sqrt(s))
end
```